lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -617,6 +619,55 @@ public void forceDeleteTask(String taskId)
     }
   }
 
+  @Override
+  public BulkOperationResults<String, TaskanaException> setOwnerOfTasks(
+      String owner, List<String> argTaskIds) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "entry to setOwnerOfTasks(owner = {}, tasks = {})",
+          owner,
+          LoggerUtils.listToString(argTaskIds));
+    }
+    try {
+      taskanaEngine.openConnection();
+      BulkOperationResults<String, TaskanaException> bulkLog = new BulkOperationResults<>();
+      if (argTaskIds == null || argTaskIds.isEmpty()) {
+        return bulkLog;
+      }
+      // remove duplicates
+      List<String> taskIds = argTaskIds.stream().sorted().distinct().collect(Collectors.toList());
+      final int requestSize = taskIds.size();
+      // use only elements we are authorized for
+      taskIds = filterForAuthorized(taskIds, bulkLog);
+      // set the Owner of these tasks we are authorized for
+      if (taskIds.isEmpty()) {
+        return bulkLog;
+      } else {
+        final int numberOfAffectedTasks = taskMapper.setOwnerOfTasks(owner, taskIds, Instant.now());
+        // check the outcome
+        List<MinimalTaskSummary> existingMinimalTaskSummaries =
```

Comment:
Cannot we do the authorization check withon the retrieval of the tasks?

Reply:
We can, but then we cannot distinguish the reason why an update failed - due to non-existence or due to missing authorization.